digital_asset_types/src/dapi/change_logs.rs

```diff
@@ -61,48 +74,169 @@ pub async fn get_proof_for_asset(
     query.sql = query
         .sql
         .replace("SELECT", "SELECT DISTINCT ON (cl_items.node_idx)");
+    let required_nodes: Vec<SimpleChangeLog> = db.query_all(query).await.map(|qr| {
+        qr.iter()
+            .map(|q| SimpleChangeLog::from_query_result(q, "").unwrap())
+            .collect()
+    })?;
+    let asset_proof = build_asset_proof(
+        leaf.tree,
+        leaf.node_idx,
+        leaf.hash,
+        &req_indexes,
+        &required_nodes,
+    );
+    Ok(asset_proof)
+}
+
+pub async fn get_asset_proof_batch(
```

Comment:
Same comment as the one on `get_asset_batch`. We have been using an extra `s` to indicate methods that return multiple items. Therefore, we should consider naming this `get_asset_proofs`, assuming the `AssetProof` is an entity, so the method will return a list of `AssetProof`s.